### ISSUE TITLE
sequence bias correction (--bias) fails with paired-end data

### ISSUE THREAD

**Issue by jaquol** [pachterlab/kallisto#102]
I am running "kallisto quant" (version 0.42.4 for Linux) on 2 toy datasets:

- (a) 1,000 paired-end reads
- (b) 1,000 single-end reads (essentially, just the first read of each pair in dataset (a))

For the (a) set the code is:
```
kallisto quant -i kallisto_homo_sapiens_hg38_gencode_v24.index -o paired_end --bias -t 8 -b 5 test_read1.fastq.gz test_read2.fastq.gz
```

For the (b) set the code is:
```
kallisto quant -i kallisto_homo_sapiens_hg38_gencode_v24.index -o single_end --bias -t 8 -b 5 --single -l 150 -s 30 test_read1.fastq.gz
```
For the latter it runs without problems. For the paired-end one however I get the following errror:
```
*** glibc detected *** /software/mb/bin/kallisto: double free or corruption (out): 0x000000001d405370 ***
======= Backtrace: =========
/lib64/libc.so.6[0x3e3c475e66]
/lib64/libc.so.6[0x3e3c4789b3]
/software/mb/bin/kallisto[0x44a404]
/software/mb/bin/kallisto[0x43f42a]
/lib64/libc.so.6(__libc_start_main+0xfd)[0x3e3c41ed5d]
/software/mb/bin/kallisto[0x440e69]
======= Memory map: ========
00400000-00730000 r-xp 00000000 00:16 2912615925                         /nfs/software/mb/el6.3/kallisto_linux-v0.42.4/kallisto
0092f000-0093a000 rw-p 0032f000 00:16 2912615925                         /nfs/software/mb/el6.3/kallisto_linux-v0.42.4/kallisto
0093a000-00960000 rw-p 00000000 00:00 0 
014fd000-414b6000 rw-p 00000000 00:00 0                                  [heap]
3e3c000000-3e3c020000 r-xp 00000000 08:03 132042                         /lib64/ld-2.12.so
3e3c21f000-3e3c220000 r--p 0001f000 08:03 132042                         /lib64/ld-2.12.so
3e3c220000-3e3c221000 rw-p 00020000 08:03 132042                         /lib64/ld-2.12.so
3e3c221000-3e3c222000 rw-p 00000000 00:00 0 
3e3c400000-3e3c58a000 r-xp 00000000 08:03 134228                         /lib64/libc-2.12.so
3e3c58a000-3e3c78a000 ---p 0018a000 08:03 134228                         /lib64/libc-2.12.so
3e3c78a000-3e3c78e000 r--p 0018a000 08:03 134228                         /lib64/libc-2.12.so
3e3c78e000-3e3c78f000 rw-p 0018e000 08:03 134228                         /lib64/libc-2.12.so
3e3c78f000-3e3c794000 rw-p 00000000 00:00 0 
3e3c800000-3e3c883000 r-xp 00000000 08:03 134967                         /lib64/libm-2.12.so
3e3c883000-3e3ca82000 ---p 00083000 08:03 134967                         /lib64/libm-2.12.so
3e3ca82000-3e3ca83000 r--p 00082000 08:03 134967                         /lib64/libm-2.12.so
3e3ca83000-3e3ca84000 rw-p 00083000 08:03 134967                         /lib64/libm-2.12.so
3e3cc00000-3e3cc17000 r-xp 00000000 08:03 135099                         /lib64/libpthread-2.12.so
3e3cc17000-3e3ce17000 ---p 00017000 08:03 135099                         /lib64/libpthread-2.12.so
3e3ce17000-3e3ce18000 r--p 00017000 08:03 135099                         /lib64/libpthread-2.12.so
3e3ce18000-3e3ce19000 rw-p 00018000 08:03 135099                         /lib64/libpthread-2.12.so
3e3ce19000-3e3ce1d000 rw-p 00000000 00:00 0 
3e3d000000-3e3d002000 r-xp 00000000 08:03 135100                         /lib64/libdl-2.12.so
3e3d002000-3e3d202000 ---p 00002000 08:03 135100                         /lib64/libdl-2.12.so
3e3d202000-3e3d203000 r--p 00002000 08:03 135100                         /lib64/libdl-2.12.so
3e3d203000-3e3d204000 rw-p 00003000 08:03 135100                         /lib64/libdl-2.12.so
3e3d800000-3e3d807000 r-xp 00000000 08:03 135121                         /lib64/librt-2.12.so
3e3d807000-3e3da06000 ---p 00007000 08:03 135121                         /lib64/librt-2.12.so
3e3da06000-3e3da07000 r--p 00006000 08:03 135121                         /lib64/librt-2.12.so
3e3da07000-3e3da08000 rw-p 00007000 08:03 135121                         /lib64/librt-2.12.so
7fa64c000000-7fa64db5c000 rw-p 00000000 00:00 0 
7fa64db5c000-7fa650000000 ---p 00000000 00:00 0 
7fa654000000-7fa655b5c000 rw-p 00000000 00:00 0 
7fa655b5c000-7fa658000000 ---p 00000000 00:00 0 
7fa658000000-7fa659b5c000 rw-p 00000000 00:00 0 
7fa659b5c000-7fa65c000000 ---p 00000000 00:00 0 
7fa65c000000-7fa65db5c000 rw-p 00000000 00:00 0 
7fa65db5c000-7fa660000000 ---p 00000000 00:00 0 
7fa660000000-7fa661b5c000 rw-p 00000000 00:00 0 
7fa661b5c000-7fa664000000 ---p 00000000 00:00 0 
7fa664000000-7fa668000000 rw-p 00000000 00:00 0 
7fa668000000-7fa66c000000 rw-p 00000000 00:00 0 
7fa66c000000-7fa670000000 rw-p 00000000 00:00 0 
7fa674000000-7fa678000000 rw-p 00000000 00:00 0 
7fa6796e1000-7fa67bbf9000 rw-p 00000000 00:00 0 
7fa67c5fa000-7fa67cdfb000 rw-p 00000000 00:00 0 
7fa67ebfe000-7fa67ebff000 ---p 00000000 00:00 0 
7fa67ebff000-7fa67f5ff000 rw-p 00000000 00:00 0 
7fa67f5ff000-7fa67f600000 ---p 00000000 00:00 0 
7fa67f600000-7fa680000000 rw-p 00000000 00:00 0 
7fa680000000-7fa684000000 rw-p 00000000 00:00 0 
7fa684028000-7fa684029000 ---p 00000000 00:00 0 
7fa684029000-7fa684a29000 rw-p 00000000 00:00 0 
7fa68542a000-7fa689df5000 rw-p 00000000 00:00 0 
7fa68a0c1000-7fa68a591000 rw-p 00000000 00:00 0 
7fa68b81d000-7fa74b964000 rw-p 00000000 00:00 0 
7fa74b964000-7fa74b97a000 r-xp 00000000 00:16 1829235194                 /nfs/software/mb/el6.3/gcc-5.1.0/build/lib64/libgcc_s.so.1
7fa74b97a000-7fa74bb79000 ---p 00016000 00:16 1829235194                 /nfs/software/mb/el6.3/gcc-5.1.0/build/lib64/libgcc_s.so.1
7fa74bb79000-7fa74bb7a000 rw-p 00015000 00:16 1829235194                 /nfs/software/mb/el6.3/gcc-5.1.0/build/lib64/libgcc_s.so.1
7fa74bb7a000-7fa74bb7c000 rw-p 00000000 00:00 0 
7fa74bb9e000-7fa74bba0000 rw-p 00000000 00:00 0 
7fff027ba000-7fff027d0000 rw-p 00000000 00:00 0                          [stack]
7fff027ff000-7fff02800000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
/users/GR/mb/jquilez/misc/rnaseq/job_cmd/test_2016_02_24_full_rnaseq-16.02.sh: line 442: 55614 Aborted
```

**Comment by pimentel** [comment]
Hi @jaquol. Thanks for the bug report. Could you privately share the data with us? Also, how does it do with only 1 thread?

Thanks!

**Comment by pmelsted** [comment]
Hi @jaquol

We are aware of this bug and have a fix working in the development version.
It will be included in the next release.

Thanks

**Comment by jaquol** [comment]
Hi @pimentel and @pmelsted,

Thanks for  your answer. 

Considering that you are already aware, do you still want me to share the data?

Regarding to the performance with only 1 thread, what metrics would you like me to know?

Cheers,

**Comment by pmelsted** [comment]
This has been resolved in version v0.42.5
